pkgs/by-name/af/affine/package.nix

```
@@ -0,0 +1,79 @@
+{ lib
+, writeText
+, fetchurl
+, stdenvNoCC
+, copyDesktopItems
+, makeDesktopItem
+, unzip
+, bash
+, electron
+, commandLineArgs ? ""
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: let
+  icon = fetchurl {
+    url = "https://raw.githubusercontent.com/toeverything/AFFiNE/v${finalAttrs.version}/packages/frontend/core/public/favicon-192.png";
+    hash = "sha256-smZ5W7fy3TK3bvjwV4i71j2lVmKSZcyhMhcWfPxNnN4=";
+  };
+  script = writeText "affine" ''
+    #!${bash}/bin/bash
+    set -e
+    export ELECTRON_IS_DEV=0
+    export ELECTRON_FORCE_IS_PACKAGED=true
+    export NODE_ENV=production
+    cd @out@
+    if [[ $EUID -ne 0 ]] || [[ $ELECTRON_RUN_AS_NODE ]]; then
+        exec ${electron}/bin/electron @out@/lib/app.asar ''${NIXOS_OZONE_WL:+''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}} ${commandLineArgs} "$@"
+    else
+        exec ${electron}/bin/electron @out@/lib/app.asar ''${NIXOS_OZONE_WL:+''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}} ${commandLineArgs} --no-sandbox "$@"
+    fi
+  '';
+in {
+  pname = "affine";
+  version = "0.13.1";
+  src = fetchurl {
+    url = "https://github.com/toeverything/AFFiNE/releases/download/v${finalAttrs.version}/affine-${finalAttrs.version}-stable-linux-x64.zip";
+    hash = "sha256-2Du5g/I82iTr8Bwb+qkLzyfbk1OrOlXqx6FHImVoAoE=";
+  };
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+  sourceRoot = ".";
+  unpackPhase = ''
+    mkdir $out
+    ${unzip}/bin/unzip $src -d $out
```

```suggestion
    unzip $src -d $out
```
unzip should go to nativeBuildInputs